kong/tools/database_cache.lua

```diff
@@ -120,21 +121,42 @@ function _M.all_apis_by_dict_key()
 end
 
 function _M.get_or_set(key, cb)
+  local lock = resty_lock:new("cache_locks", {
+    exptime = 10,
+    timeout = 10
+  })
```

Comment:
shouldn't the lock be create on the module level instead of on each call to `get_or_set`?